homeassistant/components/weatherkit/weather.py

```diff
@@ -0,0 +1,255 @@
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a weather entity from a config_entry."""
+    coordinator: WeatherKitDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    async_add_entities(
+        [
+            WeatherKitWeather(
+                coordinator,
+                config_entry.data,
+            ),
+        ]
+    )
+
+
+condition_code_to_hass = {
+    "BlowingDust": "windy",
+    "Clear": "sunny",
+    "Cloudy": "cloudy",
+    "Foggy": "fog",
+    "Haze": "fog",
+    "MostlyClear": "sunny",
+    "MostlyCloudy": "cloudy",
+    "PartlyCloudy": "partlycloudy",
+    "Smoky": "fog",
+    "Breezy": "windy",
+    "Windy": "windy",
+    "Drizzle": "rainy",
+    "HeavyRain": "pouring",
+    "IsolatedThunderstorms": "lightning",
+    "Rain": "rainy",
+    "SunShowers": "rainy",
+    "ScatteredThunderstorms": "lightning",
+    "StrongStorms": "lightning",
+    "Thunderstorms": "lightning",
+    "Frigid": "snowy",
+    "Hail": "hail",
+    "Hot": "sunny",
+    "Flurries": "snowy",
+    "Sleet": "snowy",
+    "Snow": "snowy",
+    "SunFlurries": "snowy",
+    "WintryMix": "snowy",
+    "Blizzard": "snowy",
+    "BlowingSnow": "snowy",
+    "FreezingDrizzle": "snowy-rainy",
+    "FreezingRain": "snowy-rainy",
+    "HeavySnow": "snowy",
+    "Hurricane": "exceptional",
+    "TropicalStorm": "exceptional",
+}
+
+
+def _map_daily_forecast(forecast) -> Forecast:
+    return {
+        "datetime": forecast.get("forecastStart"),
+        "condition": condition_code_to_hass[forecast.get("conditionCode")],
+        "native_temperature": forecast.get("temperatureMax"),
+        "native_templow": forecast.get("temperatureMin"),
+        "native_precipitation": forecast.get("precipitationAmount"),
+        "precipitation_probability": forecast.get("precipitationChance") * 100,
+        "uv_index": forecast.get("maxUvIndex"),
+    }
+
+
+def _map_hourly_forecast(forecast) -> Forecast:
+    return {
+        "datetime": forecast.get("forecastStart"),
+        "condition": condition_code_to_hass[forecast.get("conditionCode")],
+        "native_temperature": forecast.get("temperature"),
+        "native_apparent_temperature": forecast.get("temperatureApparent"),
+        "native_dew_point": forecast.get("temperatureDewPoint"),
+        "native_pressure": forecast.get("pressure"),
+        "native_wind_gust_speed": forecast.get("windGust"),
+        "native_wind_speed": forecast.get("windSpeed"),
+        "wind_bearing": forecast.get("windDirection"),
+        "humidity": forecast.get("humidity") * 100,
+        "native_precipitation": forecast.get("precipitationAmount"),
+        "precipitation_probability": forecast.get("precipitationChance") * 100,
+        "cloud_coverage": forecast.get("cloudCover") * 100,
+        "uv_index": forecast.get("uvIndex"),
+    }
+
+
+class WeatherKitWeather(
```

Comment:
New integrations must use `has_entity_name = True`. My advice would be to add `_attr_has_entity_name = True`, add a device (service) (without a name) and then set `_attr_name = None`. Check IPMA for an example. This way the name of the config entry is used as name for the entity, so if someone changes the config entry name, they automatically get their weather renamed

Reply:
https://developers.home-assistant.io/docs/core/entity?_highlight=has_en#has_entity_name-true-mandatory-for-new-integrations

Reply:
So, I set `_attr_name` to `None` then attached a device with `_attr_device_info`, but the sensor only gets renamed when the service (device) gets renamed, not the config entry. This is totally fine IMO, but I was wondering if I was missing something to make the config entry name match the service (and thus, the sensor's) name

Reply:
Euh, afaik that should be the case I will do some tests. For now, your code is good :)